builtin/providers/aws/validators_test.go

```diff
@@ -43,3 +43,30 @@ func TestValidateEcrRepositoryName(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateCloudWatchEventRuleName(t *testing.T) {
+	validNames := []string{
+		"HelloWorl_d",
+		"hello-world",
+		"hello.World0125",
+	}
+	for _, v := range validNames {
+		_, errors := validateCloudWatchEventRuleName(v, "name")
+		if len(errors) != 0 {
+			t.Fatalf("%q should be a valid CW event rule name: %q", v, errors)
+		}
+	}
+
+	invalidNames := []string{
+		"special@character",
+		"slash/in-the-middle",
+		// Length > 64
+		"TooLooooooooooooooooooooooooooooooooooooooooooooooooooooooongName",
```

Comment:
I'm not entirely sure it's a good idea in this context. I wouldn't want to turn this test into test for invalid characters (which may come out from `RandString`) - it would still fail (i.e. test would pass) but for a different reason. I want to be testing length only.

I'd be cautious about using `RandStringFromCharSet` in tests like this too. I believe we should be always making sure that tests (and failures too) are always well reproducible. Any randomness goes against that idea.

Repeating the same character would be a different story, but I don't think that's really necessary for 65 chars long name anyway. It fits on the screen and under 80 chars per line. :wink: 

I will keep this in mind for longer strings though.